### PR TITLE
Fix sdl does not restore subscription status on waypoint related data

### LINKED ISSUE #2445: SDL doesn't resume status of subscribtion on wayPoint-related data
### Bug Report
SDL doesn't resume status of subscribtion on wayPoint-related data

### Preconditions:
1. Into sdl_preloaded_pt.json add additional information about SubscribeWayPoints and UnSubscribeWayPoints
2. SDL and HMI are started.
3. App registered and activated.

##### Reproduction Steps
1. Do SubscribeWayPoints for this app
2. Do Unexpected disconnect via CloseMobileSession() and receive UnSubscribeWayPoints in this way
3. Register app with the same ID once more

##### Expected Behavior
Register an application successfully and resume status of subscribtion on wayPoint-related data, also resume HMIlevel being before unexpected disconnect (it means "FULL" level) 

Note: For reproduce this defect use attach script.

##### Observed Behavior
App registered with the same ID, but SDL doesn't resume status of subscribtion on wayPoint-related data .

### Attachments:
[APPLINK-33392.zip](https://github.com/smartdevicelink/sdl_core/files/2240441/APPLINK-33392.zip)

##### OS & Version Information
* OS/Version:
* SDL Core Version: 
* Testing Against:

### PR DESCRIPTION
**Pull request by mked-luxoft**
Fixes #2445 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
ATF test scripts are attached to issue

### Summary

- Added MessageHelper method that constructs message on WayPoints Subscription

- Added sending mentioned message to HMI on WayPoint Subscription added

### CLA
- [ ] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/include/application_manager/message_helper.h [hunk 1]
```diff
@@ -571,6 +571,13 @@ class MessageHelper {
    */
   static bool SendUnsubscribedWayPoints(ApplicationManager& app_mngr);
 
+  /**
+   * @brief Creates SubscribeWayPointsMessage to be sent to HMI
+   * @return filled smart object with relevant message data
+   */
+  static smart_objects::SmartObjectSPtr CreateSubscribeWayPointsMessageToHMI(
```
Comment (KVGrygoriev) [inline src/components/application_manager/include/application_manager/message_helper.h]: @mked-luxoft I'd prefer make more abstract function.
We have next function below
```
static smart_objects::SmartObjectSPtr CreateMessageForHMI(
hmi_apis::messageType::eType message_type, const uint32_t correlation_id);
```
If I we're you I'd make
```
static smart_objects::SmartObjectSPtr CreateMessageForHMI(
hmi_apis::FunctionID::eType function_id, const uint32_t correlation_id);
```

**Comment by AByzhynar** [comment]
@mked-luxoft Please make the following actions: 
- fix merge conflicts. 
- Squash commits to avoid senseless commit names

File: src/components/application_manager/include/application_manager/message_helper.h (hunk 1 above)
Comment (mked-luxoft) [reply src/components/application_manager/include/application_manager/message_helper.h]: @KVGrygoriev  fixed in a6ab85ca0ac52449928ea83373d788e6c7b429c2

File: src/components/application_manager/test/mock_message_helper.cc [hunk 2]
```diff
@@ -471,17 +477,17 @@ void MessageHelper::SendLaunchApp(const uint32_t connection_key,
       connection_key, urlSchema, packageName, app_man);
 }
 
+void MessageHelper::SendQueryApps(const uint32_t connection_key,
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/mock_message_helper.cc]: @mked-luxoft what is this method for?

**Pull request merged**